<?xml version="1.0"?>
<launch>
  <arg name="mynteye" default="mynteye" />

  <!-- node params -->

  <arg name="left_topic" default="left" />
  <arg name="right_topic" default="right" />
  <arg name="imu_topic" default="imu" />

  <arg name="base_frame_id" default="$(arg mynteye)_link" />
  <arg name="left_frame_id" default="$(arg mynteye)_left_frame" />
  <arg name="right_frame_id" default="$(arg mynteye)_right_frame" />
  <arg name="imu_frame_id" default="$(arg mynteye)_imu_frame" />

  <arg name="gravity" default="9.8" />

  <!-- device options, -1 will not set the value -->

  <!-- gain range: [0,48]-->
  <arg name="gain" default="-1" />
  <!-- <arg name="gain" default="24" /> -->

  <!-- brightness range: [0,240]-->
  <arg name="brightness" default="-1" />
  <!-- <arg name="brightness" default="120" /> -->

  <!-- contrast range: [0,255]-->
  <arg name="contrast" default="-1" />
  <!-- <arg name="contrast" default="127" /> -->

  <!-- frame_rate range: [10,60]-->
  <arg name="frame_rate" default="-1" />
  <!-- <arg name="frame_rate" default="25" /> -->

  <!-- imu_frequency range: [200,500]-->
  <arg name="imu_frequency" default="-1" />
  <!-- <arg name="imu_frequency" default="200" /> -->

  <!-- exposure_mode, 0: auto-exposure, 1: manual-exposure -->
  <arg name="exposure_mode" default="-1" />
  <!-- <arg name="exposure_mode" default="0" /> -->

  <!-- max_gain range: [0,48]-->
  <arg name="max_gain" default="-1" />
  <!-- <arg name="max_gain" default="48" /> -->

  <!-- max_exposure_time range: [0,240]-->
  <arg name="max_exposure_time" default="-1" />
  <!-- <arg name="max_exposure_time" default="240" /> -->

  <!-- desired_brightness range: [0,255]-->
  <arg name="desired_brightness" default="-1" />
  <!-- <arg name="desired_brightness" default="192" /> -->

  <!-- ir_control range: [0,160]-->
  <arg name="ir_control" default="-1" />
  <!-- <arg name="ir_control" default="0" /> -->

  <!-- hdr_mode, 0: 10-bit, 1: 12-bit -->
  <arg name="hdr_mode" default="-1" />
  <!-- <arg name="hdr_mode" default="0" /> -->

  <!-- Push down all topics/nodelets into "mynteye" namespace -->
  <group ns="$(arg mynteye)">

    <!-- mynteye_wrapper_node -->
    <node name="mynteye_wrapper_node" pkg="mynt_eye_ros_wrapper" type="mynteye_wrapper_node" output="screen">

      <!-- node params -->

      <param name="left_topic" value="$(arg left_topic)" />
      <param name="right_topic" value="$(arg right_topic)" />
      <param name="imu_topic" value="$(arg imu_topic)" />

      <param name="base_frame_id" value="$(arg base_frame_id)" />
      <param name="left_frame_id" value="$(arg left_frame_id)" />
      <param name="right_frame_id" value="$(arg right_frame_id)" />
      <param name="imu_frame_id" value="$(arg imu_frame_id)" />

      <param name="gravity" value="$(arg gravity)" />

      <!-- device options -->

      <param name="gain" value="$(arg gain)" />
      <param name="brightness" value="$(arg brightness)" />
      <param name="contrast" value="$(arg contrast)" />
      <param name="frame_rate" value="$(arg frame_rate)" />
      <param name="imu_frequency" value="$(arg imu_frequency)" />
      <param name="exposure_mode" value="$(arg exposure_mode)" />
      <param name="max_gain" value="$(arg max_gain)" />
      <param name="max_exposure_time" value="$(arg max_exposure_time)" />
      <param name="desired_brightness" value="$(arg desired_brightness)" />
      <param name="ir_control" value="$(arg ir_control)" />
      <param name="hdr_mode" value="$(arg hdr_mode)" />

    </node>

  </group> <!-- mynteye -->
</launch>
